src/cpp/logger/UserFile.cpp

```diff
@@ -0,0 +1,67 @@
+//
+// Created by s20217 on 20/04/2021.
+//
+
+#include <algorithm>
+#include <iomanip>
+#include <sstream>
+#include <list>
+
+#include "logger/UserFile.h"
+
+namespace xpansion {
+    namespace logger {
+
+        UserFile::UserFile(const std::string& filename)
+        {
+            _file.open(filename);
```

Comment:
There will be an exception raised if the file can't be open. Maybe we should add a try catch